src/hooks/use-load-data-pages.ts

```diff
@@ -46,19 +51,18 @@ const useLoadDataPages = (model: Model, layoutService: LayoutService, viewServic
       setDataPages(layoutService.layout.snapshotData?.content?.qPivotDataPages || []);
     } else if (
       (model as EngineAPI.IGenericObject)?.getHyperCubePivotData &&
-      (shouldFetchAdditionalData(qLastExpandedPos, viewService) || isMissingLayoutData(layoutService))
+      (shouldFetchAdditionalData(qLastExpandedPos, viewService) || isMissingLayoutData(layoutService, pageInfo))
     ) {
       try {
-        const pivotPages = await (model as EngineAPI.IGenericObject).getHyperCubePivotData(Q_PATH, [
-          {
-            qLeft: qLastExpandedPos ? viewService.gridColumnStartIndex : 0,
-            qTop: qLastExpandedPos ? viewService.gridRowStartIndex : 0,
-            qWidth: !viewService.gridWidth ? DEFAULT_PAGE_SIZE : viewService.gridWidth,
-            qHeight: !viewService.gridHeight ? DEFAULT_PAGE_SIZE : viewService.gridHeight,
-          },
-        ]);
+        const fetchArea: EngineAPI.INxPage = {
+          qLeft: qLastExpandedPos ? viewService.gridColumnStartIndex : 0,
+          qTop: qLastExpandedPos ? viewService.gridRowStartIndex : pageInfo.currentPage * pageInfo.rowsPerPage,
+          qWidth: !viewService.gridWidth ? DEFAULT_PAGE_SIZE : viewService.gridWidth,
+          qHeight: !viewService.gridHeight ? DEFAULT_PAGE_SIZE : viewService.gridHeight,
+        };
+        const pivotPages = await (model as EngineAPI.IGenericObject).getHyperCubePivotData(Q_PATH, [fetchArea]);
 
-        setDataPages([...qHyperCube.qPivotDataPages, ...pivotPages]);
+        setDataPages(pivotPages);
```

Comment:
@cbt1  I remember you mentioned why we need to spread previous `qPivotDataPages`, but i dont remember it clearly, but will only including what we receive from websocket cover that case you mentioned before?

Reply:
On `main` we don't really need it. But it does not hurt either, as the more values we have in our initial data the better I'd say.

In this branch my first thought is that it will no longer work. Since when the user have changed page, the initial data pages from the layout are from the first page. That would not match the current page.